### PR TITLE
feat: upgrade exoplayer to v2.17.1

### PR DESCRIPTION
**Pull request by rhysle**
#### Describe the changes
Upgrade ExoPlayer to version 2.17.1

#### Provide an example of how to test the change
Tested with a forked of [react-native-video-test](https://github.com/tailt94/react-native-video-test/tree/tailt/test)

### CODE REVIEW

**Comment by zchwyng** [comment]
Has anyone tested this?

**Comment by kesha-antonov** [comment]
Interested too

**Comment by harshaliitr** [comment]
I tested this with react native version 0.65.1. Working fine except seeking, seeking behavior is slow.

**Comment by hueniverse** [comment]
@tailt94 Can you update this to use 2.17.1 so that we jump directly to the current latest version? I promise a quick review!

**Comment by rhysle** [comment]
@hueniverse sure, but it may take some time since there's a lot of changes from v2.15.1 to v2.17.1

**Comment by hueniverse** [comment]
@tailt94 Do you think it is worth waiting? Is there enough value in moving to 2.15.1 first? I am trying to find the best way to move forward. Do you have any idea of how long it will take you to update the PR?

And again, much appreciated!

**Comment by rhysle** [comment]
@hueniverse sorry for late response. About whether it is worth or not to upgrade, I really can't answer that since I've read changelogs in exoplayer repository, there are a lot of changes since 2.15.1 and I haven't touched Android native for a long time. Even back then when I upgraded exoplayer for this pull request, it took me a lot of time because their changelog did not cover all the changes and I had to search through many places like stackoverflow to update this successfully. Maybe I'm gonna spare some time for this on this weekend.

**Comment by hueniverse** [comment]
@tailt94 That would be great. Let me know if you were able to find the time. I promise a quick review and merge this time! I would love to include this in a 6.0.0-alphax release so we get as much community feedback as possible.

**Comment by rhysle** [comment]
@hueniverse I've updated most of deprecated ones except this line and I didn't figured out how to update this, any suggestion? This method is deprecated and now replaced with [getCurrentTracksInfo](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Player.html#getCurrentTracksInfo())
https://github.com/react-native-video/react-native-video/blob/3dd73ba97e1b18ab9767a90b116681be91caab64/android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java#L191

**Comment by hueniverse** [comment]
Thanks @tailt94!

@cobarx Can you help review this and @tailt94 open question? I would really like to move to the most recent version of the player.

**Comment by sridhard** [comment]
@hueniverse @tailt94 What is the plan to merge this feature? Exoplayer solves many defects in 2.17 and we want to release our app with this merge.

**Comment by hueniverse** [comment]
@tailt94 We finally got the big Android PR merge done. As soon as you resolve the conflicts this will be the next PR to get merged! I promise!

**Comment by armands-malejevs** [comment]
I fixed the merge conflicts and tested in the example app and playback appears to work for all stream types.

**Review by armands-malejevs — APPROVED** [review]

**Comment by hueniverse** [comment]
Thanks @armadilio3! This is great. As soon as we get two more approvals we'll get v6 published.

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java
Comment (cobarx) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java:146]: What happened where we no longer need these listeners?

Comment (rhysle) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java:146]: `VideoListener` and `TextOutput` are deprecated and their methods was merged to `Player.Listener` instead. For `TextOutput` it is `onCues` method and for `VideoListener` it is `VideoListener`

**Review by cobarx — APPROVED** [review]

**Comment by sridhard** [comment]
@hueniverse @cobarx any update on this?

**Pull request merged**
